Src/AutoFakeItEasy/FakeItEasyMethodQuery.cs

```diff
@@ -74,8 +74,46 @@ public IEnumerable<ParameterInfo> Parameters
 
             public object Invoke(IEnumerable<object> parameters)
             {
-                return new Fake<T>(
-                    b => b.WithArgumentsForConstructor(parameters));
+                var genericFakeType = typeof (Fake<>).MakeGenericType(typeof (T));

```

Comment:
No change would be required to AutoFixture.AutoFakeItEasy2. It works out of the box.
I was suggesting the change to AutoFixture.AutoFakeItEasy so as to spare the users from "nuget package whiplash".

> wouldn't it be easier

I suppose it depends on your focus. You have a working PR in your hands, so the initial development is done, without the need to explain to users that they should go to AutoFakeItEasy2. That being said, the reflection-based approach I've committed is reflection-based, and therefore somewhat less desirable. I understand your concerns about it. Personally, I'd take the PR (easy for me to say; I wrote it) in order to provide the users with a seamless experience.
I don't think there will ever be a need to revisit the method in question during FakeItEasy 2.x's run (and probably not for a while beyond that), so maintenance should be nil.

Of course that same argument applies to the duplicated code if you decide to go with reinstating AutoFakeItEasy2. Me, I think the internal costs and benefits are so close that I'd let the decision come down to what's going to work for the users. But I know that it's you that has to live with whatever decision you make.